sunpy/net/dataretriever/sources/fermi_gbm.py

```diff
@@ -0,0 +1,125 @@
+from urllib.parse import urljoin
+
+from ..client import GenericClient
+
+__all__ = ['GBMClient']
+
+
+class GBMClient(GenericClient):
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+
+        Returns the url for Fermi/GBM data for the given date.
+
+        baseurl = 'https://heasarc.gsfc.nasa.gov/FTP/fermi/data/gbm/daily/'
+
+
+        Parameters
+        ----------
+        timerange : sunpy.time.TimeRange
+            time range for which to download the data
+
+        Returns
+        -------
+        url for time of interest
+        """
+
+        # Checks if detector keyword
+        # If not defaults to detector 5
+        if 'detector' in kwargs:
+            det = _check_detector(kwargs['detector'])
+
+        else:
+            det = 'n5'
+
+        # Check for datatype keyword - either CSPEC or CTIME
+        # Default type is CSPEC
+
+        if 'datatype' in kwargs:
+            data_type = _check_type(kwargs['datatype'])
+        else:
+            data_type = 'cspec'
+
+        days = timerange.get_dates()
+        urls = []
+        for day in days:
+            urls.append(self._get_url_for_date(day, det, data_type, **kwargs))
+        return urls
+
+    def _get_url_for_date(self, date, det, data_type,  **kwargs):
+        """
+        Returns URL dats of interest
+
+        Parameters
+        ----------
+        date : 'datetime.date'
+
+        Returns
+        -------
+        url : str
+            the url at the date of interest
+
+        """
+        filename = 'glg_' + data_type + '_' + det + date.strftime('_%y%m%d_') + 'v00.pha'
```

Comment:
I appreciate this is the code from lightcurve, and I am happy to merge it as-is.

If you want some magic bonus points, you could investigate switching this over to our wonderful [`sunpy.util.scraper.Scraper`](https://docs.sunpy.org/en/latest/api/sunpy.util.scraper.Scraper.html#sunpy.util.scraper.Scraper) class which is designed to facilitate this kind of URL templateing.

Reply:
Magic bonus points! now using sunpy.util.scraper.Scraper instead of _get_url_for_date